resources/Components/LazyLoadImage/index.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState, useCallback } from 'react'
+
+export const LazyLoadImage = ({ src, placeholder, loadError, ...props }) => {
+  const [imgSrc, initImg] = useState(placeholder || src)
+  const onLoad = useCallback(() => {
+    initImg(src)
+  }, [src])
+  const onError = useCallback(() => {
+    initImg(loadError || placeholder)
+  }, [loadError, placeholder])
+  useEffect(() => {
+    const imageObj = new Image()
+    imageObj.src = src
+    imageObj.addEventListener('load', onLoad)
+    imageObj.addEventListener('error', onError)
+    return () => {
+      imageObj.removeEventListener('load', onLoad)
+      imageObj.removeEventListener('error', onError)
+    }
+  }, [src, onLoad, onError])
```

Comment:
do we want to run this effect when these functions change? seems like we only want to run it when the `src` prop changes